common/scala/src/main/scala/org/apache/openwhisk/core/entity/WhiskAction.scala

```diff
@@ -55,7 +55,8 @@ case class WhiskActionPut(exec: Option[Exec] = None,
                           limits: Option[ActionLimitsOption] = None,
                           version: Option[SemVer] = None,
                           publish: Option[Boolean] = None,
-                          annotations: Option[Parameters] = None) {
+                          annotations: Option[Parameters] = None,
+                          unlock: Option[Boolean] = None) {
```

Comment:
concretely my suggestion is to change `unlock: Option[Boolean]` to `permissions: Option[Int]` and use bit operations to toggle permissions. The first set of permissions to respect is whether the resource is locked locked or unlocked. This can be abstracted with utility methods.

Reply:
I already changed to `unix permission style` to `protect action`